File: go/count-of-matches-in-tournament.go
func algorithmSolution(n int) int {
	matchCount := 0
	for 1 < n {
		isOdd := n & 1
		matchInRound := n / 2
		matchCount += matchInRound
		n = matchInRound + isOdd
	}
	return matchCount
}

func numberOfMatches(n int) int {
	//return algorithmSolution(n)
	return n - 1
}

File: python/count-of-matches-in-tournament.py
class Solution:
    def algorithm(n: int) -> int:
        match_count = 0
        while 1 < n:
            is_odd = n & 1
            match_in_round = n // 2
            match_count += match_in_round
            n = match_in_round + is_odd
        return match_count

    def numberOfMatches(self, n: int) -> int:
        # return algorithm(n)
        return n - 1
